README.md

```diff
@@ -381,6 +387,52 @@ Sometimes you may want to use your own serialization for some objects and do not
 
 MyGlobalSerializer is your implementation of StreamSerializer interface. You do not need to write the getHazelcastTypeId function for your objects which are handled by MyGlobalSerializer class.
 
+# Invocation And Retry Mechanism
+The client works by sending requests to the members in the cluster and processes the reponses. This is called a client invocation to the cluster. If any problem occurs during an invocation, the following rules apply to decide if the invocation should be retried:
+- If the problem is an IOException, which means that the request could not even be sent to the cluster member (e.g. due to a network connection problem), then all such requests can be retried.
+- If the invocation were sent to the cluster and the member returned an exception to the client, the client will retry the invocation only if the operation is a retryable idempotent operation. For example, the IMap::get is idempotent and can be retried but IMap::put is not idempotent.
+
+When invocation is being retried, the client may wait some time before it retries again, this retry wait time is configurable with this configuration property:
+- config.setProperty("hazelcast.client.invocation.retry.pause.millis", "500");
+The default retry wait time is 1000 msecs(i.e. 1 second).
+
+Of course, this retry does not continue indefinitely but finishes when the invocation timeout occurs. This duration is configurable with this configuration property:
+- config.setProperty("hazelcast.client.invocation.timeout.seconds", "30");
+The default timeout is 120 seconds.
+
+## Client Backpressure
+Sometimes, you may want to slow down the client operations to the cluster (e.g. your servers are overloaded). Then the client can be configured to wait until number of outstanding invocations whose reponses are not received to become less than a certain number. This is called "Client Backpressure". By default, the backpressure is disabled. There are a few properties which control the backpressure. These client configuration properties are:
```

Comment:
Sometimes, e.g., when your servers are overloaded, you may want to slow down the client operations to the cluster.